From e0236515d23be44996bffdfc239c831d1c2dc4e2 Mon Sep 17 00:00:00 2001
From: Sandor Yu <Sandor.yu@nxp.com>
Date: Fri, 18 May 2018 15:18:09 +0800
Subject: [PATCH 3852/5242] MLK-18355-3: HDP API: Add new API function
 CDN_API_GetClock

commit  18b062dc1fe14542d398ecd6168500e493f807ef from
https://source.codeaurora.org/external/imx/linux-imx.git

Add new API function CDN_API_GetClock.

Signed-off-by: Sandor Yu <Sandor.yu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mxc/hdp/API_General.c |    5 +++++
 drivers/mxc/hdp/API_General.h |    2 ++
 2 files changed, 7 insertions(+)

diff --git a/drivers/mxc/hdp/API_General.c b/drivers/mxc/hdp/API_General.c
index 3bec3c5..85c0b61 100644
--- a/drivers/mxc/hdp/API_General.c
+++ b/drivers/mxc/hdp/API_General.c
@@ -361,6 +361,11 @@ CDN_API_STATUS CDN_API_SetClock(state_struct *state, u8 MHz)
 	return cdn_apb_write(state, SW_CLK_H << 2, MHz);
 }
 
+CDN_API_STATUS CDN_API_GetClock(state_struct *state, u32 *MHz)
+{
+	return cdn_apb_read(state, SW_CLK_H << 2, MHz);
+}
+
 CDN_API_STATUS CDN_API_General_Read_Register(state_struct *state, u32 addr,
 					     GENERAL_Read_Register_response *resp)
 {
diff --git a/drivers/mxc/hdp/API_General.h b/drivers/mxc/hdp/API_General.h
index f41c332..1a17a20 100644
--- a/drivers/mxc/hdp/API_General.h
+++ b/drivers/mxc/hdp/API_General.h
@@ -224,6 +224,8 @@ CDN_API_STATUS CDN_API_ApbConf_blocking(state_struct *state, u8 dpcd_bus_sel,
  */
 CDN_API_STATUS CDN_API_SetClock(state_struct *state, u8 MHz);
 
+CDN_API_STATUS CDN_API_GetClock(state_struct *state, u32 *MHz);
+
 CDN_API_STATUS CDN_API_General_Read_Register(state_struct *state, u32 addr,
 					     GENERAL_Read_Register_response *resp);
 CDN_API_STATUS CDN_API_General_Read_Register_blocking(state_struct *state,
-- 
1.7.9.5

